main/modes/mode_flight.c

```diff
@@ -0,0 +1,1374 @@
+          int (*compar)(const void *, const void *));
+int abs(int j);
+#endif
+
+/*============================================================================
+ * Variables
+ *==========================================================================*/
+
+swadgeMode modeFlight =
+{
+    .modeName = "Flyin Donut",
+    .fnEnterMode = flightEnterMode,
+    .fnExitMode = flightExitMode,
+    .fnButtonCallback = flightButtonCallback,
+    .fnBackgroundDrawCallback = flightBackground,
+    .wifiMode = NO_WIFI,
+    .fnEspNowRecvCb = NULL,
+    .fnEspNowSendCb = NULL,
+    .fnMainLoop = flightRender,
+    .fnAccelerometerCallback = NULL,
+    .fnAudioCallback = NULL,
+};
+flight_t* flight;
+
+static const char fl_title[]  = "Flightsim";
+static const char fl_flight_env[] = "Take Flight";
+static const char fl_flight_invertY0_env[] = "Y NOT INVERTED";
+static const char fl_flight_invertY1_env[] = "Y INVERTED";
+static const char fl_flight_perf[] = "Perf Test";
+static const char str_quit[] = "Quit";
+static const char str_high_scores[] = "High Scores";
+
+
+
+/*============================================================================
+ * Functions
+ *==========================================================================*/
+
+static void flightBackground(display_t* disp, int16_t x, int16_t y, int16_t w, int16_t h, int16_t up, int16_t upNum )
+{
+    fillDisplayArea( disp, x, y, x+w, h+y, CNDRAW_BLACK );
+}
+
+/**
+ * Initializer for flight
+ */
+static void flightEnterMode(display_t * disp)
+{
+    // Alloc and clear everything
+    flight = malloc(sizeof(flight_t));
+    memset(flight, 0, sizeof(flight_t));
+
+    // Hmm this seems not to be obeyed, at least not well?
+    setFrameRateUs( 33333 );
+
+    flight->mode = FLIGHT_MENU;
+    flight->disp = disp;
+
+    const uint16_t * data = model3d;//(uint16_t*)getAsset( "3denv.obj", &retlen );
+    data+=2; //header
+    flight->enviromodels = *(data++);
+    flight->environment = malloc( sizeof(const tdModel *) * flight->enviromodels );
+    int i;
+    for( i = 0; i < flight->enviromodels; i++ )
+    {
+        const tdModel * m = flight->environment[i] = (const tdModel*)data;
+        data += 8 + m->nrvertnums + m->nrfaces * m->indices_per_face;
+    }
+
+    loadFont("ibm_vga8.font", &flight->ibm);
+    loadFont("radiostars.font", &flight->radiostars);
+
+    loadFont("mm.font", &flight->meleeMenuFont);
+    flight->menu = initMeleeMenu(fl_title, &flight->meleeMenuFont, flightMenuCb);
+    addRowToMeleeMenu(flight->menu, fl_flight_env);
+    addRowToMeleeMenu(flight->menu, fl_flight_perf);
+    flight->invertYRow = addRowToMeleeMenu( flight->menu, flight->inverty?fl_flight_invertY1_env:fl_flight_invertY0_env );
+    addRowToMeleeMenu(flight->menu, str_quit);
+    addRowToMeleeMenu(flight->menu, str_high_scores);
+}
+
+/**
+ * Called when flight is exited
+ */
+static void flightExitMode(void)
+{
+    deinitMeleeMenu(flight->menu);
+    freeFont(&flight->meleeMenuFont);
```

Comment:
Loading three fonts, but freeing one? Looks like someone could use a `valgrind`.

Reply:
Whoops.  On windows rn.

Reply:
https://drmemory.org/ is the closest I've found to valgrind on windows